pdns/zoneparser-tng.cc

```diff
@@ -272,6 +272,38 @@ pair<string,int> ZoneParserTNG::getLineNumAndFile()
     return {d_filestates.top().d_filename, d_filestates.top().d_lineno};
 }
 
+void fixTxtRecord(string &txt) {
+  bool in_quote = false;
+  bool needs_end_quote = false;
+
+  // Bind automatically puts quotes around text blocks that don't have them
+  for( string::size_type i = 0; i < txt.size(); ++i ) {
+    if( in_quote ) {
```

Comment:
I think this will break on unseparated \<character-string>s like `TXT foo"bar"`, though to be honest I'm not sure what other software does with those either. But does seem to accept e.g. `TXT "foo"bar` and `TXT "foo""bar"`, so there's an asymmetry if nothing else.